### PR TITLE
fix(sns): allow default SNS policy with SourceOwner

### PR DESCRIPTION
**Pull request by christiandavilakoobin**
### Context

Prowler alerts for sns public accessibilty checks on topics that are generated from the AWS console.

### Description

The AWS console uses the not very well documented AWS:SourceOwner attribute on defaults policies: https://github.com/awsdocs/iam-user-guide/issues/111

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/cloudfront/cloudfront_distributions_https_enabled/cloudfront_distributions_https_enabled.py
Comment (jfagoagas) [inline prowler/providers/aws/services/cloudfront/cloudfront_distributions_https_enabled/cloudfront_distributions_https_enabled.py:1]: Why is this file in the PR? I think this issue with CloudFront was fixed a long ago. Please remove it if it is out of the SNS scope.

File: tests/providers/aws/services/cloudfront/cloudfront_distributions_https_enabled/cloudfront_distributions_https_enabled_test.py
Comment (jfagoagas) [inline tests/providers/aws/services/cloudfront/cloudfront_distributions_https_enabled/cloudfront_distributions_https_enabled_test.py:1]: Why is this file in the PR? I think this issue with CloudFront was fixed a long ago. Please remove it if it is out of the SNS scope.

File: prowler/providers/aws/lib/policy_condition_parser/policy_condition_parser.py [hunk 1]
```diff
@@ -6,12 +6,14 @@ def is_account_only_allowed_in_condition(
     valid_condition_options = {
         "StringEquals": [
             "aws:SourceAccount",
+            "aws:SourceOwner",
```
Comment (jfagoagas) [inline prowler/providers/aws/lib/policy_condition_parser/policy_condition_parser.py]: Can you please update the library tests?

**Review by jfagoagas — CHANGES_REQUESTED** [review]
That's great, please check the comments to merge it.

Thanks!

File: prowler/providers/aws/services/cloudfront/cloudfront_distributions_https_enabled/cloudfront_distributions_https_enabled.py
Comment (christiandavilakoobin) [reply prowler/providers/aws/services/cloudfront/cloudfront_distributions_https_enabled/cloudfront_distributions_https_enabled.py:1]: Sorry, I had a modified repo, and I didn't update it correctly. I pass it clean on PR https://github.com/prowler-cloud/prowler/pull/2698

Comment (jfagoagas) [reply prowler/providers/aws/services/cloudfront/cloudfront_distributions_https_enabled/cloudfront_distributions_https_enabled.py:1]: Good, also we have to fix the cloudfront type issue. PR incoming.

**Pull request closed without merging**